trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/FunctionHandler.java

```diff
@@ -716,6 +688,57 @@ private Result handleFunctionCallGivenNameAndArguments(final IDispatcher main, f
 			translatedParams.add(in.getLrValue().getValue());
 			functionCallExpressionResultBuilder.addAllExceptLrValue(in);
 		}
+		if (calleeProcCType != null && calleeProcCType.hasVarArgs()) {
+			final boolean hasUsedVarargs = calleeProcCType.getVarArgsUsage() == VarArgsUsage.USED;
+			// For varargs we need a special handling:
+			// - If the varargs are not used, we simply dispatch the arguments, without passing them to the function.
+			// - If they are used, we create a pointer for all the remaining arguments and pass them to the function.
+			final AuxVarInfo auxvarinfo = mAuxVarInfoBuilder.constructAuxVarInfo(loc,
+					mTypeHandler.constructPointerType(loc), SFO.AUXVAR.VARARGS_POINTER);

```

Comment:
I think you'll probably need to call some allocation function to get the pointer. If you simply use a nondeterministic pointer, wouldn't that

- possibly overwrite the contents of any other pointer?
- violate memsafety?

Reply:
Also, I'm not sure  what the current effect is -- if the call appears in the loop, does using ther auxVar (without e.g. havocing it) not mean that the same pointer is used in every iteration? Is this intended, or can it have harmful effects?

Reply:
Oh wait, I see you initialize the pointer below. My bad.